src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Concatenate.cs

```diff
@@ -132,9 +126,16 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
 
             // Type check the args
             for (var i = 0; i < count; i++)
-            {
-                fArgsValid &= CheckParamIsTypeOrSingleColumnTable(DType.String, args[i], argTypes[i], errors, out var isTable, ref nodeToCoercedTypeMap);
-                hasTableArg |= isTable;
+            {
+                if (argTypes[i].IsTable && argTypes[i] != DType.ObjNull)

```

Comment:
> argTypes[i] != DType.ObjNull [](http://example.com/codeflow?start=42&length=29)

Should this check be done in the Check<Type>ColumnType (i.e., return true, with no coercions, if arg is DType.ObjNull)?

Reply:
As discussed, we are inconsistent on how untyped null tables are handled.  I added a comment and some tests to illustrate the point.